target/linux/ath79/files/drivers/mtd/nand/rb4xx_nand.c

```diff
@@ -0,0 +1,269 @@
+// SPDX-License-Identifier: GPL-2.0-only
+/*
+ * NAND flash driver for the MikroTik RouterBoard 4xx series
+ *
+ * Copyright (C) 2008-2011 Gabor Juhos <juhosg@openwrt.org>
+ * Copyright (C) 2008 Imre Kaloz <kaloz@openwrt.org>
+ * Copyright (C) 2015 Bert Vermeulen <bert@biot.com>
+ *
+ * This file was based on the driver for Linux 2.6.22 published by
+ * MikroTik for their RouterBoard 4xx series devices.
+ *
+ * The NAND chip containing the root filesystem is behind an SPI-connected
+ * CPLD. The CPLD also acts as a GPIO expander, and the ALE/CLE/NCE pins are
+ * wired to the CPLD.
+ */
+
+#include <linux/mtd/rawnand.h>
+#include <linux/spi/spi.h>
+#include <linux/platform_device.h>
+#include <linux/gpio/driver.h>
+#include <linux/gpio/consumer.h>
+#include <linux/module.h>
+#include <linux/of.h>
+
+#include <asm/mach-ath79/rb4xx_cpld.h>
```

Comment:
The include should be located in a place consistent with driver location within the kernel tree: for instance with a driver sitting in `drivers/spi/` the corresponding include belongs to `include/linux/spi/`